game/world/triggers/abstracttrigger.cpp

```diff
@@ -53,10 +54,16 @@ bool AbstractTrigger::isEnabled() const {
   }
 
 void AbstractTrigger::processEvent(const TriggerEvent& evt) {
-  if(emitTimeLast>0 && world.tickCount()<emitTimeLast+uint64_t(fireDelaySec*1000.f)) {
+  if(emitTimeLast>0 && world.tickCount()<emitTimeLast+uint64_t(reTriggerDelaySec*1000.f))
+    return;
+  if(fireWaitTime==0)
```

Comment:
how it should work, if for example `retriggerWaitSec==1`, and `fireDelaySec==10` and triger been activated multiple times, before firing? Should we have multiple events in flight to cover that case?

`TriggerEvent::timeBarrier` seem to be a good candidate to implement fire-delay mechanics

Reply:
> TriggerEvent::timeBarrier seem to be a good candidate 

`if(t.name()==e.target) { // NOTE: trigger name is not unique - more then one trigger can be activated`

I played around a bit but this gives me a headache. To prevent infinite trigger cycles and assign wait time `triggerevent` has to reference trigger directly instead of name. This would require rewrite of constructor and adjust fucntion calls that pass trigger events. Just asking if you see a simpler answer or is this the way to go?